soroban-auth/src/testutils.rs

```diff
@@ -38,6 +38,25 @@ pub mod ed25519 {
         (signer.identifier(env), signer)
     }
 
+    /// Create a signer from a ed25519 secret key.
+    pub fn signer(
+        env: &Env,
+        secret_key: &[u8; 32],
+    ) -> (
+        IdentifierValue,
+        impl Identifier + Sign<SignaturePayload, Signature = [u8; 64]> + Debug,
+    ) {
+        let secret = ed25519_dalek::SecretKey::from_bytes(secret_key).unwrap();
+        let public = ed25519_dalek::PublicKey::from(&secret);
+        let signer = ed25519_dalek::Keypair { secret, public };
```

Comment:
@heytdep I renamed the `build` function to `signer` because I really like this idea of being able to construct the types, and it seems useful to side-by-side add a `identifier` function to get an identifier from the same type of input.

I changed the inputs of `build` to be a single secret key because the public key can be derived from the secret key. That way in the test you only have to provide one parameter, not two, which is a little more convenient. This also ensures the public key and secret key aren't out of sync, because there's no validation they are a match.